module_4/hw1.py

```diff
@@ -0,0 +1,88 @@
+"""
+Create classes to track homeworks.
+
+1. Homework - accepts howework text and deadline (datetime.timedelta)   -   Done
+Homework has a method, that tells if deadline has passed.
+
+2. Student - can solve homework with `do_homework` method.              -   Done
+Raises DeadlineError with "You are late" message if deadline has passed
+
+3. Teacher - can create homework with `create_homework`; check homework with `check_homework`.
+Any teacher can create or check any homework (even if it was created by one of colleagues).
+
+Homework are cached in dict-like structure named `homework_done`. Key is homework, values are 
+solutions. Each student can only have one homework solution.
+
+Teacher can `reset_results` - with argument it will reset results for specific homework, without - 
+it clears the cache.
+
+Homework is solved if solution has more than 5 symbols.
+
+-------------------
+Check file with tests to see how all these classes are used. You can create any additional classes 
+you want.
+"""
+import datetime
+
+
+class Homework:
+    def __init__(self, homework_text, deadline: datetime.timedelta.days):
+        self.homework_text = homework_text
+        self.deadline = deadline
+
+    def deadline_has_passed(self) -> bool:
+        if self.deadline < 0:
```

Comment:
Deadline should be a timedelta object https://docs.python.org/3/library/datetime.html#timedelta-objects 
When we are creating hw, we can "remember" its creation date. When we check hw - we check that creation date + deadline is greater or equal then current datetime

Reply:
Deadline should be a timedelta object - done;
When we are creating hw, we can "remember" its creation date - done;
When we check hw - we check that creation date + deadline is greater or equal then current datetime - done, but
not in check_hw() but in the .do_homework() method because the test expects the exception from .do_homework():
![image](https://github.com/sergei820/PythonTAProgramEpam/assets/56273100/0b0caf21-e61b-4b76-b930-99d00b99982a)